src/main/java/uk/gov/pay/directdebit/mandate/model/Mandate.java

```diff
@@ -10,16 +10,16 @@
 
 public class Mandate {
     private final String description;
-    private Long id;
+    private final Long id;
     private final MandateExternalId externalId;
-    private MandateState state;
+    private final MandateState state;
     private final GatewayAccount gatewayAccount;
     private final String returnUrl;
-    private MandateBankStatementReference mandateBankStatementReference;
+    private final MandateBankStatementReference mandateBankStatementReference;
     private final String serviceReference;
     private final ZonedDateTime createdDate;
-    private Payer payer;
-    private PaymentProviderMandateId paymentProviderMandateId;
+    private final Payer payer;
+    private final PaymentProviderMandateId paymentProviderMandateId;
 
     private Mandate(MandateBuilder builder) {
         this.id = builder.id;
```

Comment:
Can we add some `Objects.requireNonNull(…)` magic to fields that always have to be present?

Reply:
yes, great idea, will do.